cmd/kube-controller-manager/app/validatingadmissionpolicystatus.go

```diff
@@ -20,32 +20,46 @@ import (
 	"context"
 
 	admissionregistrationv1alpha1 "k8s.io/api/admissionregistration/v1alpha1"
-	pluginvalidatingadmissionpolicy "k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy"
-	"k8s.io/apiserver/pkg/cel/openapi/resolver"
-	"k8s.io/client-go/kubernetes/scheme"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	externalinformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
 	"k8s.io/controller-manager/controller"
 	"k8s.io/kubernetes/pkg/controller/validatingadmissionpolicystatus"
-	"k8s.io/kubernetes/pkg/generated/openapi"
 )
 
 var validatingAdmissionPolicyResource = admissionregistrationv1alpha1.SchemeGroupVersion.WithResource("validatingadmissionpolicies")
 
 func startValidatingAdmissionPolicyStatusController(ctx context.Context, controllerContext ControllerContext) (controller.Interface, bool, error) {
+	const registeredControllerName = "validatingadmissionpolicy-status-controller"
 	// intended check against served resource but not feature gate.
 	// KCM won't start the controller without the feature gate set.
 	if !controllerContext.AvailableResources[validatingAdmissionPolicyResource] {
 		return nil, false, nil
 	}
-	typeChecker := &pluginvalidatingadmissionpolicy.TypeChecker{
-		SchemaResolver: resolver.NewDefinitionsSchemaResolver(scheme.Scheme, openapi.GetOpenAPIDefinitions),
-		RestMapper:     controllerContext.RESTMapper,
+	config := controllerContext.ClientBuilder.ConfigOrDie(registeredControllerName)
+	extensionsClient, err := apiextensions.NewForConfig(config)
+	if err != nil {
+		return nil, false, err
 	}
+	// The controllerContext does not provide an informer factory for the extensions' client, manually
+	// create the informer factory.
+	// Note that we do not bypass the factory or construct the informer manually to simplify the startup logic.
+	// This should be changed once other controllers start to make use of informers of API extensions.
+	extensionsInformerFactory := externalinformers.NewSharedInformerFactory(extensionsClient, controllerContext.ResyncPeriod())
```

Comment:
why not add this to the controllerContext now?

Reply:
Adding it into controllerContext will expand the exposure to all KCM controllers which we didn't do due to consideration of limiting the impact. I am ok with adding to controllerContext

Reply:
The original idea is to limit the scope of change. Updating to add it to controllerContext

Reply:
Updated.